src/new/sendTransaction.ts

```diff
@@ -494,4 +509,162 @@ class SendTransaction extends EventEmitter {
   }
 }
 
-export default SendTransaction;
+/**
+ * Check the tx data and propose inputs and outputs to complete the transaction.
+ * We will only check a single token
+ *
+ * @param {IStorage} storage
+ * @param {IDataTx} dataTx
+ * @param {IUtxoSelectionOptions} options
+ */
+export async function prepareSendTokensData(
+  storage: IStorage,
+  dataTx: IDataTx,
+  options: IUtxoSelectionOptions = {},
+): Promise<Pick<IDataTx, 'inputs'|'outputs'>> {
+  const token = options.token || HATHOR_TOKEN_CONFIG.uid;
+  const utxoSelection = options.utxoSelectionMethod || bestUtxoSelection;
+  const newtxData: Pick<IDataTx, 'inputs'|'outputs'> = { inputs: [], outputs: [] };
+  let outputAmount = 0;
+
+  // Calculate balance for the token on the transaction
+  for (const output of dataTx.outputs) {
+    if (isDataOutputCreateToken(output)) {
```

Comment:
With this line are you ignoring melt and mint authorities? Or should you check `isAuthorityOutput(output)`?

Reply:
The outputs provided by the user should be in the form of an `ISendOutput` which does not allow authorities.
The only other way to provide outputs would be to provide the `Transaction` instance which does not use this method.

Here we are ignoring "minted custom tokens" and "HTR melted withdrawals" in the outputs provided by the user.
This is to ensure the typescript compiler of the type of the output since `IDataOutputCreateToken` does not have a token property.